esmvalcore/preprocessor/_volume.py

```diff
@@ -275,6 +277,59 @@ def volume_statistics(cube, operator):
     return _create_cube_time(src_cube, result, times)
 
 
+def axis_statistics(cube, axis, operator):
+    """Perform statistics along a given axis.
+
+    Operates over an axis direction. If weights are required,
+    they are computed using the coordinate bounds.
+
+    Arguments
+    ---------
+    cube: iris.cube.Cube
+        Input cube.
+    axis: str
+        Direction over where to apply the operator. Possible values
+        are 'x', 'y', 'z', 't'.
+    operator: str
+        Statistics to perform. Available operators are:
+        'mean', 'median', 'std_dev', 'sum', 'variance',
+        'min', 'max', 'rms'.
+
+    Returns
+    -------
+    iris.cube.Cube
+        collapsed cube.
+    """
+    try:
+        coord = cube.coord(axis=axis)
+    except iris.exceptions.CoordinateNotFoundError as err:
+        raise ValueError(
+            'Axis {} not found in cube {}'.format(
+                axis,
+                cube.summary(shorten=True))) from err
+    coord_dims = cube.coord_dims(coord)
+    if len(coord_dims) > 1:
+        raise NotImplementedError(
+            'axis_statistics not implemented for '
+            'multidimensional coordinates.')
+    operation = get_iris_analysis_operation(operator)
+    weights = None
+    if operator_accept_weights(operator):
+        coord_dim = coord_dims[0]
+        expand = list(range(cube.ndim))
+        expand.remove(coord_dim)
+        weights = np.abs(coord.bounds[..., 1] - coord.bounds[..., 0])
```

Comment:
Let's keep it lazy:
```suggestion
        bounds = coord.core_bounds()
        weights = np.abs(bounds[..., 1] - bounds[..., 0])
```

Reply:
I cannot call `expand_dims` in dask, so the wouldn't the bounds get realised later on anyway? They are always going to be a 1D array though, so I thought it wouldn't be a big deal if they are not lazy.

Reply:
The `expand_dims` function has been added in [recent dask versions](https://docs.dask.org/en/stable/changelog.html#id45), but don't worry about it. You should be able to use `np.expand_dims` and have one of NumPy's dispatch mechanisms (e.g. [NEP-18](https://numpy.org/neps/nep-0018-array-function-protocol.html)) figure out the right implementation.